examples/advanced/bionemo/downstream/finetune_esm2.py

```diff
@@ -0,0 +1,503 @@
+        strategy=strategy,
+        limit_val_batches=limit_val_batches,  # This controls upsampling and downsampling
+        val_check_interval=val_check_interval,
+        log_every_n_steps=log_every_n_steps,
+        num_nodes=num_nodes,
+        callbacks=callbacks,
+        plugins=nl.MegatronMixedPrecision(
+            precision=precision,
+            params_dtype=get_autocast_dtype(precision),
+            pipeline_dtype=get_autocast_dtype(precision),
+            grad_reduce_in_fp32=grad_reduce_in_fp32,
+            autocast_enabled=False,
+        ),
+    )
+
+    # (2) patch the lightning trainer
+    flare.patch(trainer, restore_state=False, load_state_dict_strict=False)
+
+    # (3) receives FLModel from NVFlare
+    # Note that we don't need to pass this input_model to trainer
+    # because after flare.patch the trainer.fit/validate will get the
+    # global model internally
+    input_model = flare.receive()
+    print(
+        f"\n[Current Round={input_model.current_round}, Site = {flare.get_site_name()}, Global model = {input_model} ({len(input_model.params)} params)]\n"
+    )
+    # use a unique result directory for each round
+
+    # Remove previous checkpoints to preserve disk space
+    keep_last_ckpt_only = True  # TODO: make configurable
+    if keep_last_ckpt_only:
+        previous_ckpt_dir = result_dir / f"round{input_model.current_round-1}" / experiment_name / "dev" / "checkpoints"
+        if previous_ckpt_dir.is_dir():
+            print(f"Removing previous checkpoint directory {previous_ckpt_dir}")
+            shutil.rmtree(previous_ckpt_dir)
+
+    # create output folder for this round
+    result_dir = result_dir / f"round{input_model.current_round}"
+
+    # add a learning rate decay for each round
+    if input_model.current_round > 0:
+        lr_step_reduce = 1.05  # TODO: make lr_step_reduce configurable
+        new_lr = lr / (input_model.current_round * lr_step_reduce)
+        new_lr_multiplier = lr_multiplier / (input_model.current_round * lr_step_reduce)
+        print(f"Reduce lr {lr} by {input_model.current_round*lr_step_reduce}: {new_lr}")
+    else:
+        new_lr = lr
+        new_lr_multiplier = lr_multiplier
+
+    # remaining bionemo training code
+    tokenizer = get_tokenizer()
+
+    # Initialize the data module.
+    train_dataset = dataset_class.from_csv(train_data_path, task_type=task_type, label_column=label_column)
+    valid_dataset = dataset_class.from_csv(valid_data_path, task_type=task_type, label_column=label_column)
+    if task_type == "classification":
+        if classes:
+            if not isinstance(classes, List):
+                raise ValueError(f"classes is expected to be list of strings but received {type(classes)}: {classes}")
+            train_dataset.label_tokenizer.build_vocab([classes])
+            print(f"Build custom label tokenizer based on label classes: {classes}")
+        valid_dataset.label_tokenizer = train_dataset.label_tokenizer
```

Comment:
I moved this to the `ESM2FineTuneDataModule` [here](https://github.com/NVIDIA/bionemo-framework/pull/689/files#diff-cb96e7f2feb5fc55ef7a97d0ef3baefcc4f311562fbbfbd2cc30173119f71e27R131-R134). You can remove it from here.

Reply:
Thanks. Removed `valid_dataset.label_tokenizer = train_dataset.label_tokenizer`. We still need to rebuild the label tokenizer for training, though as we need to make sure that the same one is used on each client regardless of their dataset distributions.